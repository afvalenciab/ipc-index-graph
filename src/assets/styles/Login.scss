$font-Nunito: 'Nunito Sans', Arial;
$color-font: #000;

%flex-settings {
  display: flex;
  justify-content: center;
  align-items: center;
}

%a-settings {
  text-decoration: none;
  color: #006AFA;
  font-size: 0.9em;
  transition: 0.1s;
  will-change: transform;
}
%a-transform {
  text-decoration: underline;
  transform: scale(1.02);
  transform-origin: 0 100%;
}

.container__login {
  display: grid;
  grid-template: minmax(auto, 1fr) / 1fr;
  grid-gap: 10px;
  min-height: 100vh;
  color: #000;
  padding: 40px;
  box-sizing: border-box;
  align-content: center;
  justify-items: center;
}

.login {
  display: grid;
  grid-template-rows: repeat(3, auto);
  grid-row-gap: 5px;
  align-items: center;
  align-content: space-evenly;
  font-size: 16px;
  border: 1px solid $color-font;
  border-radius: 25px;
  padding: 0 30px;
  height: 100%;
  width: 320px;
  box-sizing: border-box;

  h2 {
    margin: 8px 0;
    font-size: 70px;
    font-weight: 900;
    line-height: .9;
    letter-spacing: 3px;
  }

  form {
    > input {
      display: block;
      width: 100%;
      box-sizing: border-box;
      margin-bottom: 15px;
      outline: none;
      border: 0;
      border-bottom: 1px solid $color-font;
      background: transparent;
      padding: 10px;
      font-size: 0.9em;
      color: $color-font;
      font-family: $font-Nunito;
    }
    input::placeholder {
      color: rgba(0,0,0,.6);
      font-weight: 100;
    }

    button {
      width: 100%;
      border: 0;
      height: 40px;
      border-radius: 50px;
      outline: none;
      font-family: $font-Nunito;
      font-size: 0.9em;
      letter-spacing: 1px;
      background: #006AFA;
      color: #FFF;
      margin-bottom: 15px;
      cursor: pointer;
      transition: 0.3s;
      will-change: transform;
    }
    button:active {
      transform: scale(0.9);
    }
  }

  .login__singup {
    font-size: 0.9em;
    p {
      margin: 0;
      margin-bottom: 15px;
      color: rgba(0,0,0,.7);
    }
    a {
      display: inline-block;
      @extend %a-settings;
      font-size: 1em;
      font-weight: 600;
      padding: 0 5px;
    }
    a:hover {
      @extend %a-transform;
    }
  }
}

@media screen and (max-width: 480px ) {
  .container__login {
    padding: 0px;
  }
  
  .login {
    font-size: 15px;
    border: 0px;
    padding: 10px 30px;
    width: 100%;
    box-sizing: border-box;
  
    h2 {
      font-size: 50px;
    }
  }
}