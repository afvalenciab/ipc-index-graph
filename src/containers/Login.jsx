import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { loginRequest, loginUser } from '../actions';
import MessageError from '../components/MessageError';
import '../assets/styles/Login.scss';

const Login = (props) => {
  const { loginRequest, loginUser, user } = props;
  const [form, setForm] = useState({
    email: '',
  });

  const handleInput = (event) => {
    setForm({
      ...form,
      [event.target.name]: event.target.value,
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    loginRequest(form);
  };

  const handleChangePage = () => {
    loginUser({});
  };

  useEffect(() => {
    if (Object.keys(user).length > 0) {
      props.history.push('/');
    };
  }, [user]);

  return (
    <main className='container__login'>
      <section className='login'>
        <h2>Inicia sesión</h2>
        <form onSubmit={handleSubmit}>
          <input name='email' type='email' placeholder='Correo' onChange={handleInput} required />
          <input name='password' type='Password' placeholder='Constraseña' onChange={handleInput} required />
          <button type='submit'>Iniciar sesión</button>
          {!user && <MessageError />}
        </form>
        <div className='login__social'>
          <a href='/login'>
            Iniciar sesión con Google
          </a>
        </div>
        <div className='login__singup'>
          <p>
            ¿Aún no estás registrado?
            <Link to='/register' onClick={handleChangePage}>
              Regístrate
            </Link>
          </p>
        </div>
      </section>
    </main>
  );
};

const mapStateToProps = (state) => {
  return {
    user: state.user,
  };
};

const mapDispatchToProps = {
  loginRequest,
  loginUser,
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
