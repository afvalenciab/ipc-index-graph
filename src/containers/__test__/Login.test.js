import React from 'react';
import { create } from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProviderMock from '../../__mocks__/ProviderMocks';
import Login from '../Login';

configure({ adapter: new Adapter() });

describe('Login testing', () => {
  test('Match Snapshot', () => {
    const login = create(
      <ProviderMock>
        <Login />
      </ProviderMock>
    );
    expect(login.toJSON()).toMatchSnapshot();
  });

  test('Calls and execute preventDefault function onSubmit form', () => {
    const preventDefault = jest.fn();
    const login = mount(
      <ProviderMock>
        <Login />
      </ProviderMock>
    );
    login.find('form').simulate('submit', { preventDefault });
    expect(preventDefault).toHaveBeenCalledTimes(1);
    login.unmount();
  });
});
