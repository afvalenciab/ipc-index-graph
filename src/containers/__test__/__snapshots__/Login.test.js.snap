// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login testing Match Snapshot 1`] = `
<main
  className="container__login"
>
  <section
    className="login"
  >
    <h2>
      Inicia sesión
    </h2>
    <form
      onSubmit={[Function]}
    >
      <input
        name="email"
        onChange={[Function]}
        placeholder="Correo"
        required={true}
        type="email"
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="Constraseña"
        required={true}
        type="Password"
      />
      <button
        type="submit"
      >
        Iniciar sesión
      </button>
    </form>
    <div
      className="login__singup"
    >
      <p>
        ¿Aún no estás registrado?
        <a
          href="/register"
          onClick={[Function]}
        >
          Regístrate
        </a>
      </p>
    </div>
  </section>
</main>
`;
